file_example.c

```diff
@@ -97,8 +97,8 @@ file_handler_run(void *arg)
 
 		/* process command */
 		result = file_handle_cmd(h->dev, cmd);
-		pthread_mutex_lock(&state->completion_mtx);
 		tcmulib_command_complete(h->dev, cmd, result);
+		pthread_mutex_lock(&state->completion_mtx);
 		tcmulib_processing_complete(h->dev);
 		pthread_mutex_unlock(&state->completion_mtx);
```

Comment:
It's related to the bug, but with larger scale testing this patch does not fix all the problems, so hold off on merging please. 

The problem I now see is if you create multiple mappings and then delete the mappings it will cause tcmu-runner to hang which needs a restart to be able to recover. The key thing that causes that breakage is the deleting of multiple mappings... If you delete one at a time and exit targetcli each time it is fine...